packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -222,10 +222,17 @@ export class TableLayoutManager<TData extends TableRecord> {
     }
 
     private resetGridSizedColumns(): void {
-        if (!this.gridSizedColumns) {
+        if (!this.gridSizedColumns?.length) {
             return;
         }
 
+        let gridSizeColumnTotal = 0;
+        for (const gridSizedColumn of this.gridSizedColumns) {
+            gridSizeColumnTotal
+                += gridSizedColumn.columnInternals.currentPixelWidth!;
+        }
```

Comment:
nit: I think we should prefer array methods over loops, but that's a preference not a coding convention. Something like this (you should check my work).
```suggestion
        const gridSizeColumnTotal = this.gridSizedColumns.reduce((sum, col) => sum + col.columnInternals.currentPixelWidth!, 0);
```